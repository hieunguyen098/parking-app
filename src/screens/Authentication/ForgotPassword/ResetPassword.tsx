import { Text, View } from 'react-native';
import React from 'react';
import LargeButton from '../../../components/Buttons/LargeButton';
import { useNavigation } from '@react-navigation/native';
import PasswordInput from '../../../components/PasswordInput/PasswordInput';
import styles from '../styles';

const ResetPassword = () => {
    const navigation: any = useNavigation();
    const navigate = (name: string) => {
        navigation.navigate(name);
    };
    return (
        <View style={styles.container}>
            <View style={styles.group}>
                <Text style={styles.description}>Nhập mật khẩu của bạn</Text>
                <PasswordInput length={6} />
                <Text style={styles.description}>Nhập lại mật khẩu</Text>
                <PasswordInput length={6} />
            </View>
            <LargeButton onPress={() => navigate('Login')} title="Tiếp tục" style={styles.continueButton} />
        </View>
    );
};

export default ResetPassword;
