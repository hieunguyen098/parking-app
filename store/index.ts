import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './slices/authSlice';
import { generalReducer } from './slices/generalSlice';

export const store = configureStore({
    reducer: {
        auth: authReducer,
        general: generalReducer,
    },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
