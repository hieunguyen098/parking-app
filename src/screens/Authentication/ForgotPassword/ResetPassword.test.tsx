import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { store } from '../../../../store';
import ResetPassword from './ResetPassword';

jest.useFakeTimers();
test('test password', () => {
    const queryClient = new QueryClient();
    const component = (
        <QueryClientProvider client={queryClient}>
            <Provider store={store}>
                <NavigationContainer>
                    <ResetPassword />
                </NavigationContainer>
            </Provider>
        </QueryClientProvider>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
