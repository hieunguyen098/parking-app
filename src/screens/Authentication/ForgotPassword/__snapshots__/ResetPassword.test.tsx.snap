// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test password 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Nhập mật khẩu của bạn
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Array [
            Object {
              "backgroundColor": "#FBFBFB",
              "borderColor": "#FBFBFB",
              "borderRadius": 20,
              "borderWidth": 1,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 30,
            },
            false,
            false,
          ],
        ]
      }
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <TextInput
        autoCapitalize="characters"
        autoCorrect={false}
        blurOnSubmit={false}
        caretHidden={true}
        clearButtonMode="never"
        disableFullscreenUI={true}
        keyboardType="number-pad"
        maxLength={6}
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onPressIn={[Function]}
        spellCheck={false}
        style={
          Object {
            "bottom": 0,
            "fontSize": 1,
            "left": 0,
            "opacity": 0.01,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Nhập lại mật khẩu
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Array [
            Object {
              "backgroundColor": "#FBFBFB",
              "borderColor": "#FBFBFB",
              "borderRadius": 20,
              "borderWidth": 1,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 30,
            },
            false,
            false,
          ],
        ]
      }
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(217, 217, 217, 1)",
            "borderRadius": 100,
            "fontSize": 0,
            "height": 10,
            "lineHeight": 5,
            "marginHorizontal": 8,
            "textAlign": "center",
            "width": 10,
          }
        }
      />
      <TextInput
        autoCapitalize="characters"
        autoCorrect={false}
        blurOnSubmit={false}
        caretHidden={true}
        clearButtonMode="never"
        disableFullscreenUI={true}
        keyboardType="number-pad"
        maxLength={6}
        onBlur={[Function]}
        onChange={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onPressIn={[Function]}
        spellCheck={false}
        style={
          Object {
            "bottom": 0,
            "fontSize": 1,
            "left": 0,
            "opacity": 0.01,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#FF953A",
          "borderRadius": 8,
          "paddingHorizontal": 12,
          "paddingVertical": 18,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#FF953A",
        },
        Object {
          "borderRadius": 100,
          "marginBottom": 24,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "600",
            "textAlign": "center",
          },
          Object {
            "color": "#fff",
          },
          undefined,
        ]
      }
    >
      Tiếp tục
    </Text>
  </View>
</View>
`;
